src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/EditingTargetInlineChunkExecution.java

```diff
@@ -0,0 +1,184 @@
+               outputs_.remove(output.chunkId());
+            }
+            else
+            {
+               // we already have an output panel for this input, and it's
+               // either waiting to execute or currently executing. ignore this
+               // request to re-execute the range, as it's likely to be an
+               // unintended duplicate.
+               return;
+            }
+         }
+      }
+      
+      // create dummy scope for execution
+      Scope scope = Scope.createRScopeNode(
+            chunkId,
+            range.getStart(),
+            range.getEnd(),
+            Scope.SCOPE_TYPE_CHUNK);
+      
+      // create popup panel to host output
+      final ChunkInlineOutput output = new ChunkInlineOutput(chunkId, range);
+      output.addCloseHandler(new CloseHandler<PopupPanel>()
+      {
+         @Override
+         public void onClose(CloseEvent<PopupPanel> event)
+         {
+            outputs_.remove(chunkId);
+         }
+      });
+
+      // render offscreen until complete
+      output.setPopupPosition(-100000, -100000);
+      output.show();
+      outputs_.put(chunkId, output);
+
+      SendToChunkConsoleEvent event =
+            new SendToChunkConsoleEvent(docId_, scope, range, 
+                  NotebookQueueUnit.EXEC_SCOPE_INLINE);
+      events_.fireEvent(event);
+   }
+   
+   // Handlers ----
+   
+   @Override
+   public void onConsoleWriteError(ConsoleWriteErrorEvent event)
+   {
+      if (outputs_.containsKey(event.getConsole()))
+        outputs_.get(event.getConsole()).onConsoleWriteError(event);
+   }
+
+   @Override
+   public void onConsoleWriteOutput(ConsoleWriteOutputEvent event)
+   {
+      if (outputs_.containsKey(event.getConsole()))
+        outputs_.get(event.getConsole()).onConsoleWriteOutput(event);
+   }
+
+   @Override
+   public void onChunkExecStateChanged(ChunkExecStateChangedEvent event)
+   {
+      // ignore if not targeted at one of our chunks
+      if (event.getDocId() != docId_ || 
+          !outputs_.containsKey(event.getChunkId()))
+         return;
+      
+      ChunkInlineOutput output = outputs_.get(event.getChunkId());
+      
+      if (event.getExecState() == NotebookDocQueue.CHUNK_EXEC_STARTED)
+      {
+         output.setState(ChunkInlineOutput.State.Started);
+      }
+      else if (event.getExecState() == NotebookDocQueue.CHUNK_EXEC_FINISHED)
+      {
+         output.setState(ChunkInlineOutput.State.Finished);
+         final ScreenCoordinates pos = computePopupPosition(
+               output.range(), output.getOffsetWidth());
+         output.setPopupPosition(pos.getPageX(), pos.getPageY());
+         output.show();
+      }
+   }
+
+   private ScreenCoordinates computePopupPosition(Range range, int width)
+   {
+      Rectangle bounds = display_.getRangeBounds(range);
+      Point center = bounds.center();
+      
+      int pageX = center.getX() - (width / 2);
+      int pageY = bounds.getBottom() + 10;
```

Comment:
That's a good idea. For extra credit I've also taken care of this for MathJax popups. :-) Change here: https://github.com/rstudio/rstudio/pull/768/commits/26a57c65bb321c4fd8ecdc8cf630183def865d8a